src/settings/cloud.ts

```diff
@@ -0,0 +1,247 @@
+
+let isServerRunning = false;
+
+async function preCheck() {
+    if (!cloudHost) {
+        console.error("Cloud host not defined");
+        return;
+    }
+
+    if (!cloudToken) {
+        startCloudServer();
+        console.error("Cloud token not defined");
+        const loginUrl = `${cloudHost}/login`;
+        await shell.openExternal((loginUrl));
+
+        while (!cloudToken) {
+            await new Promise(resolve => setTimeout(resolve, 1000));
+        }
+        return;
+    }
+}
+
+export async function deleteCloud() {
+    console.log("Deleting cloud settings.");
+
+    await preCheck();
+
+    const path = getCloudConfigLocation();
+    fs.unlinkSync(path);
+
+    const deletefetch = await fetch(`${cloudHost}/delete`, {
+        method: 'GET',
+        headers: {
+            'Content-Type': 'application/json',
+            'Authorization': `${cloudToken}`
+        }
+    });
+
+    const deletejson = await deletefetch.json();
+
+    if (deletejson.error) {
+        console.warn("Error deleting cloud settings:", deletejson.error);
+        dialog.showMessageBoxSync({
+            type: "error",
+            title: "Error deleting cloud settings",
+            message: deletejson.error + "\n\nPlease try by restarting GoofCord and try again.",
+            buttons: ["OK"]
+        });
+
+        return;
+    }
+
+    console.log("Deleted cloud settings.");
+
+    dialog.showMessageBoxSync({
+        type: "info",
+        title: "Settings deleted",
+        message: "Settings deleted successfully.",
+        buttons: ["OK"]
+    });
+
+    return;
+}
+
+export async function loadCloud() {
+    console.log("Loading cloud settings.");
+
+    await preCheck();
+
+    const loadfetch = await fetch(`${cloudHost}/load`, {
+        method: 'GET',
+        headers: {
+            'Content-Type': 'application/json',
+            'Authorization': `${cloudToken}`
+        }
+    });
+
+    const loadjsonraw = await loadfetch.json();
+
+    if (loadjsonraw.error) {
+        console.warn("Error loading cloud settings:", loadjsonraw.error);
+        dialog.showMessageBoxSync({
+            type: "error",
+            title: "Click Delete Cloud Data in the settings and try again.",
+            message: loadjsonraw.error,
+            buttons: ["OK"]
+        });
+        return;
+    }
+
+    const loadjson = JSON.stringify(loadjsonraw);
+    const loadjsonobj = JSON.parse(loadjson);
+
+    console.log("Loaded cloud settings:", loadjson);
+
+    const key = getConfig("cloudEncryptionKey");
+
+    if (!key) {
+        console.warn("Cloud Encryption Key not set.");
+        dialog.showMessageBoxSync({
+            type: "error",
+            title: "Cloud Encryption Key not set",
+            message: "Please set the Cloud Encryption Key in the settings and try again.",
+            buttons: ["OK"]
+        });
+        return;
+    }
+
+    const iv = crypto.randomBytes(16);
+
+    setConfigBulk(loadjsonobj);
```

Comment:
Does it load from cloud on every start? The cachedConfig is set to { cloudHost: 'https://goofcordcloud.wuemeli.com' } here

Reply:
No, only if the Load from Cloud Button is pressed